src/braket/circuits/circuit.py

```diff
@@ -1304,18 +1310,73 @@ def _get_frames_waveforms_from_instrs(
                     _validate_uniqueness(waveforms, waveform)
                     waveforms[waveform.id] = waveform
             # this will change with full parametric calibration support
-            elif hasattr(type(instruction.operator), "angle"):
-                if gate_calibrations is not None:
-                    key = (type(instruction.operator)(FreeParameter("theta")), instruction.target)
-                    if key in gate_calibrations:
-                        ps = gate_calibrations.get(key)
-                        bound_key = (
-                            type(instruction.operator)(instruction.operator.angle),
-                            instruction.target,
-                        )
-                        gate_calibrations[bound_key] = ps(theta=instruction.operator.angle)
+            elif (
+                isinstance(instruction.operator, Parameterizable) and gate_calibrations is not None
+            ):
+                self._add_fixed_argument_calibrations(gate_calibrations, instruction)
         return frames, waveforms
 
+    def _add_fixed_argument_calibrations(
+        self,
+        gate_calibrations: Dict[Tuple[Gate, QubitSet], PulseSequence],
+        instruction: Instruction,
+    ) -> None:
+        """Adds calibrations with arguments set to the instruction parameter values
+
+        Given the collection of parameters in instruction.operator, this function looks for matching
+        parametric calibrations that have free parameters. If such a calibration is found and the
+        number N of its free parameters equals the number of instruction parameters, we can bound
+        the arguments of the calibration and add it to the calibration dictionary.
+
+        If N is smaller, it is probably impossible to assign the instruction parameter values to the
+        corresponding calibration parameters so we raise an error.
+        If N=0, we ignore it as it will not be removed by _generate_frame_wf_defcal_declarations.
+
+        This function modifies its `gate_calibrations` input argument.
+
+        Args:
+            gate_calibrations (Dict[Tuple[Gate, QubitSet], PulseSequence]): a dictionary of
+                calibrations
+            instruction (Instruction): a Circuit instruction
+
+        Raises:
+            NotImplementedError: in two cases: (i) if the instruction contains unbound parameters
+                and the calibration dictionary contains a parametric calibration applicable to this
+                instructions; (ii) if the calibration is defined with a partial number of unbound
+                parameters.
+        """
+        for key, calibration in deepcopy(gate_calibrations).items():
+            gate = key[0]
+            target = key[1]
+            if target != instruction.target:
+                continue
+            if (
+                gate._qasm_name == instruction.operator._qasm_name
+                and isinstance(gate, Parameterizable)
```

Comment:
```suggestion
                type(gate) == type(instruction.operator)
```
We check the gate before calling the function for `Parameterizable`